components/omega/test/ocn/TendencyTermsTest.cpp

```diff
@@ -0,0 +1,676 @@
+       Geom, Mesh, OnVertex, NVertLevels);
+
+   // Compute numerical result
+   Array2DReal NumVelHyperDiff("NumVelHyperDiff", Mesh->NEdgesOwned,
+                               NVertLevels);
+
+   VelocityHyperDiffOnEdge VelHyperDiffOnE(Mesh, TendConfig);
+   parallelFor(
+       {Mesh->NEdgesOwned, NVertLevels}, KOKKOS_LAMBDA(int IEdge, int KLevel) {
+          VelHyperDiffOnE(NumVelHyperDiff, IEdge, KLevel, DivCell, RVortVertex);
+       });
+
+   // Compute errors
+   ErrorMeasures VelHyperDiffErrors;
+   Err += computeErrors(VelHyperDiffErrors, NumVelHyperDiff, ExactVelHyperDiff,
+                        Mesh, OnEdge, NVertLevels);
+
+   // Check error values
+   if (!isApprox(VelHyperDiffErrors.LInf, Setup.ExpectedLaplaceErrorLInf,
+                 RTol)) {
+      Err++;
+      LOG_ERROR("TendencyTermsTest: VelHyperDiff LInf FAIL");
+   }
+
+   if (!isApprox(VelHyperDiffErrors.L2, Setup.ExpectedLaplaceErrorL2, RTol)) {
+      Err++;
+      LOG_ERROR("TendencyTermsTest: VelHyperDiff L2 FAIL");
+   }
+
+   if (Err == 0) {
+      LOG_INFO("TendencyTermsTest: VelHyperDiff PASS");
+   }
+
+   return Err;
+} // end testVelHyperDiff
+
+int initTendTest(const std::string &mesh) {
+
+   I4 Err = 0;
+
+   MachEnv::init(MPI_COMM_WORLD);
+   MachEnv *DefEnv  = MachEnv::getDefaultEnv();
+   MPI_Comm DefComm = DefEnv->getComm();
+
+   I4 IOErr = IO::init(DefComm);
+   if (IOErr != 0) {
+      Err++;
+      LOG_ERROR("AuxVarsTest: error initializing parallel IO");
+   }
+
+   int DecompErr = Decomp::init(mesh);
+   if (DecompErr != 0) {
+      Err++;
+      LOG_ERROR("AuxVarsTest: error initializing default decomposition");
+   }
+
+   int HaloErr = Halo::init();
+   if (HaloErr != 0) {
+      Err++;
+      LOG_ERROR("AuxVarsTest: error initializing default halo");
+   }
+
+   int MeshErr = HorzMesh::init();
+   if (MeshErr != 0) {
+      Err++;
+      LOG_ERROR("AuxVarsTest: error initializing default mesh");
+   }
+
+   return Err;
+} // end initTendTest
+
+void finalizeTendTest() {
+
+   HorzMesh::clear();
+   Halo::clear();
+   Decomp::clear();
+   MachEnv::removeAll();
+
+} // end finalizeTendTest
+
+void tendencyTermsTest(const std::string &mesh = DefaultMeshFile) {
+
+   int Err = initTendTest(mesh);
+   if (Err != 0) {
+      LOG_CRITICAL("TendencyTermsTest: Error initializing");
+   }
+
+   const auto &Mesh = HorzMesh::getDefault();
+   int NVertLevels  = 16;
+
+   const Real RTol = sizeof(Real) == 4 ? 1e-2 : 1e-10;
+
+   Err += testThickFluxDiv(NVertLevels, RTol);
+
+   Err += testPotVortHAdv(NVertLevels, RTol);
+
+   Err += testKEGrad(NVertLevels, RTol);
+
+   Err += testSSHGrad(NVertLevels, RTol);
+
+   Err += testVelDiff(NVertLevels, RTol);
+
+   Err += testVelHyperDiff(NVertLevels, RTol);
+
+   if (Err == 0) {
+      LOG_INFO("TendencyTermsTest: Successful completion");
+   }
+
+   finalizeTendTest();
+
+} // end tendencyTermsTest
+
+int main(int argc, char *argv[]) {
```

Comment:
With recent changes to the testing, we require main to actually return the error code, so you'll want to have tendencyTermsTest to return Err and then main return that Err as an exit code. This was to catch certain errors that didn't create a FAIL but caused the code to exit.